app/Http/Controllers/AccountController.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Account;
+use function CsrDelft\getDateTime;
+use CsrDelft\model\entity\security\AccessRole;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+
+class AccountController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    /**
+     * @param Account|null $account
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function bewerken(Account $account = null)
+    {
+        $account = $account ?? Auth::user();
+
+        $this->authorize('view', $account);
+
+        return view('account.bewerken', ['account' => $account]);
+    }
+
+    /**
+     * @param Request $request
+     * @param Account $account
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     * @throws \Illuminate\Validation\ValidationException
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function opslaan(Request $request, Account $account = null)
+    {
+        $account = $account ?? Auth::user();
+
+        $this->authorize('update', $account);
+        $this->validator($account, $request->all())->validate();
+
+        $this->update($account, $request->all());
+
+        if ($this->shouldUpdatePassword($request)) {
+            if ($account->hasPermission('P_LEDEN_MOD') || Auth::validate($request->only(['uid', 'password']))) {
+                $this->updatePassword($account, $request->input('password_nieuw'));
+            }
+        }
+
+        return back();
+    }
+
+    private function shouldUpdatePassword(Request $request)
+    {
+        return $request->filled(['password_nieuw', 'password_nieuw_confirmation']);
+    }
+
+    protected function validator($account, $data)
+    {
+        return Validator::make($data, [
+            'username' => 'required|unique:accounts,username,' . $account->uid . ',uid|max:255',
+            'email' => 'required|unique:accounts,email,' . $account->uid . ',uid|email',
+            'perm_role' => ['required', Rule::in(AccessRole::canChangeAccessRoleTo($account->perm_role))],
```

Comment:
In plaats van '$account->perm_role' moet dit de AccessRole van de ingelogde gebruiker zijn.

Reply:
Jep